### ISSUE TITLE
False positive in Lint/UnescapedBracketInRegexp when has a bare ] as first element of a negated character class

### ISSUE THREAD

**Issue by timdiggins** [rubocop/rubocop#15307]
### Before submitting

- [x] I have searched the existing issues and this hasn't been reported yet.
- [x] I am running the latest stable version of RuboCop.

### Issue type

False positive

### Cop name

Lint/UnescapedBracketInRegexp

### Steps to reproduce the problem

Given a spec with this
```ruby
  let(:sample_regexp) { /[^]]/ }
```

### Expected behavior

No failures

### Actual behavior

rubocop response:

```
  Lint/UnescapedBracketInRegexp: Regular expression has ] without escape.
    let(:sample_regexp) { /[^]]/ }
                              ^
```

### RuboCop configuration

```yaml

```

### RuboCop version

```shell
1.88.0 (using Parser 3.3.11.1, rubocop-ast 1.49.1, analyzing as Ruby 3.1, running on ruby 3.4.8) [arm64-darwin24]
  - rubocop-capybara 2.23.0
  - rubocop-factory_bot 2.28.0
  - rubocop-performance 1.26.1
  - rubocop-rails 2.35.4
  - rubocop-rspec 3.10.2
  - rubocop-rspec_rails 2.32.0
  - rubocop-thread_safety 0.7.3
```
